sphinx/application.py

```diff
@@ -298,10 +299,18 @@ def _init_builder(self, buildername):
         if buildername is None:
             print('No builder selected, using default: html', file=self._status)
             buildername = 'html'
-        if buildername not in self.builderclasses:
-            raise SphinxError('Builder name %s not registered' % buildername)
-
-        builderclass = self.builderclasses[buildername]
+        entry_points = iter_entry_points('sphinx.builders', buildername)
+        try:
+            entry_point = next(entry_points)
+        except StopIteration:
+            # fall back to old method
+            if buildername not in self.builderclasses:
+                raise SphinxError('Builder name %s not registered'
+                                  % buildername)
+            builderclass = self.builderclasses[buildername]
+        else:
+            builderclass = entry_point.load()
+            builderclass.setup(self)
```

Comment:
I'm now using ``extenion_module.setup()``. The extension module is the module where the builder class is defined.

I have also fixed handling of built-in builders and made sure that specifying the extension in ``conf.py/extensions`` still works in case the builder does provide the entry point.